test/util/mutex.test.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+const should = require('should');
+const sinon = require('sinon');
+const Bluebird = require('bluebird');
+const mockRequire = require('mock-require');
+const {
+  errors: {
+    InternalError: KuzzleInternalError
+  }
+} = require('kuzzle-common-objects');
+
+const KuzzleMock = require('../mocks/kuzzle.mock');
+
+describe('#mutex', () => {
+  let Mutex = require('../../lib/util/mutex');
+  let kuzzle;
+
+  beforeEach(() => {
+    kuzzle = new KuzzleMock();
+  });
+
+  describe('#lock', () => {
+    it('should be able to lock a ressource', () => {
+      kuzzle.services.list.internalCache.set.resolves(true);
+
+      const mutex = new Mutex(kuzzle, 'foo', { timeout: 0, ttl: 123 });
+
+      return mutex.lock()
+        .then(result => {
+          should(result).be.true();
+          should(kuzzle.services.list.internalCache.set).calledWith(
+            'foo',
+            sinon.match.string,
+            'NX',
+            'PX',
+            123);
+        });
+    });
+
+    it('should use random values for different mutex instances', () => {
+      const mutex1 = new Mutex(kuzzle, 'foo');
+      const mutex2 = new Mutex(kuzzle, 'foo');
+
+      should(mutex1.mutexId).not.eql(mutex2.mutexId);
+    });
```

Comment:
Is this test relevant with only 2 mutex?

Reply:
IMHO it's enough to test with 2 different mutex since the goal is to ensure a random value is correctly added.

Maybe if we create one billion of mutex we may have collision but still I doubt it